dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/UserDatastoreController.java

```diff
@@ -219,18 +231,44 @@ public WebMessage updateUserValue(
     @ResponseBody
     public WebMessage deleteUserValue(
         @PathVariable String namespace,
-        @PathVariable String key )
+        @PathVariable String key,
+        @RequestParam( required = false ) String username )
     {
-        UserDatastoreEntry userEntry = userDatastoreService.getUserEntry(
-            currentUserService.getCurrentUser(), namespace, key );
+        UserDatastoreEntry entry = getEntry( username, namespace, key );
 
-        if ( userEntry == null )
+        if ( entry == null )
         {
             return notFound( "The key '" + key + "' was not found in the namespace '" + namespace + "'." );
         }
 
-        userDatastoreService.deleteUserEntry( userEntry );
+        userDatastoreService.deleteUserEntry( entry );
 
         return ok( "Key '" + key + "' deleted from the namespace '" + namespace + "'." );
     }
+
+    private UserDatastoreEntry getEntry( String username, String namespace, String key )
+    {
+        return userDatastoreService.getUserEntry( getUser( username ), namespace, key );
+    }
+
+    @Nonnull
+    private User getUser( String username )
+    {
+        User currentUser = currentUserService.getCurrentUser();
+        if ( username == null || username.isBlank() )
+        {
+            return currentUser;
+        }
+        if ( !currentUser.isSuper() )
+        {
+            throw new IllegalQueryException(
+                "Only superusers can read or write other users data using the `user` parameter." );
+        }
+        User user = userService.getUserByUsername( username );
+        if ( user == null )
+        {
+            throw new IllegalQueryException( "No user with username " + username + " does exist." );
```

Comment:
"No user with username X exists." or "There is no user with username X." might be better.

Reply:
I try to sneak in this wording change in some of my next PRs ;)

Reply:
Good plan! :)

Thanks!